Modules/xSharePoint/DSCResources/MSFT_xSPInstall/MSFT_xSPInstall.psm1

```diff
@@ -81,15 +81,27 @@ function Set-TargetResource
     
     $setup = Start-Process -FilePath $setupExe -ArgumentList "/config `"$configPath`"" -Wait -PassThru
 
-    if ($setup.ExitCode -eq 0) {
-        Write-Verbose -Message "SharePoint binary installation complete"
-        $global:DSCMachineStatus = 1
-    }
-    else
-    {
-        throw "SharePoint install failed, exit code was $($setup.ExitCode)"
+    switch ($setup.ExitCode) {
+        0 {  
+            Write-Verbose -Message "SharePoint binary installation complete"
+            $global:DSCMachineStatus = 1
+        }
+        30066 {
+            if (    ((Get-Item 'HKLM:\Software\Microsoft\Windows\CurrentVersion\Component Based Servicing\RebootPending' -ErrorAction SilentlyContinue) -ne $null) `
```

Comment:
Is there a module to provide this validation? If not, should we wrap this as a function inside .Util?